huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go

```diff
@@ -633,33 +768,41 @@ func resizeInstance(ctx context.Context, d *schema.ResourceData, meta interface{
 		return fmtp.Errorf("error creating HuaweiCloud DMS instance client: %s", err)
 	}
 
-	product, err := getProductDetail(config, d, instanceType)
-	if err != nil || product == nil {
-		return fmtp.Errorf("change storage_space failed, error querying product detail: %s", err)
-	}
-	logp.Printf("[DEBUG] Get DMS product detail: %#v", product)
+	opts := instances.ResizeInstanceOpts{}
+	if _, ok := d.GetOk("product_id"); ok {
+		product, err := getProductDetail(config, d, instanceType)
+		if err != nil || product == nil {
+			return fmtp.Errorf("change storage_space failed, error querying product detail: %s", err)
+		}
+		logp.Printf("[DEBUG] Get DMS product detail: %#v", product)
 
-	storage := d.Get("storage_space").(int)
-	specCode := product.SpecCode
-	if instanceType == "rabbitmq" {
-		specCode = product.ProductInfos[0].SpecCode
-	}
-	opts := instances.ResizeInstanceOpts{
-		NewSpecCode:     specCode,
-		NewStorageSpace: storage,
+		opts.NewStorageSpace = d.Get("storage_space").(int)
+		opts.NewSpecCode = product.SpecCode
+		if instanceType == "rabbitmq" {
```

Comment:
ERROR?  you declare the instanceType is single or cluster at the top of file

Reply:
`engineType` is better.